features/paginering.feature

```diff
@@ -0,0 +1,78 @@
+
+  Scenario: de zoekvraag levert meerdere pagina's en er wordt geen page parameter gebruikt
+    Als de request wordt gedaan naar /adressen?pandidentificatie=0826100000000467
+    Dan zitten er 20 adressen in het antwoord
+    En zijn dit adressen 1 tot en met 20 met deze pandidentificatie
+    En is attribuut _links.self.href gelijk aan "/adressen?pandidentificatie=0826100000000467"
+    En bevat het antwoord geen attribuut _links.first
+    En bevat het antwoord geen attribuut _links.previous
+    En is attribuut _links.next.href gelijk aan "/adressen?pandidentificatie=0826100000000467&page=2"
+    En is attribuut _links.self.href gelijk aan "/adressen?pandidentificatie=0826100000000467"
+
+  Scenario: de zoekvraag levert meerdere pagina's en met de page parameter wordt de eerste pagina gevraagd
+    Als de request wordt gedaan naar /adressen?pandidentificatie=0826100000000467&page=1
+    Dan zitten er 20 adressen in het antwoord
+    En zijn dit adressen 1 tot en met 20 met deze pandidentificatie
+    En is attribuut _links.self.href gelijk aan "/adressen?pandidentificatie=0826100000000467&page=1"
+    En bevat het antwoord geen attribuut _links.first
+    En bevat het antwoord geen attribuut _links.previous
+    En is attribuut _links.next.href gelijk aan "/adressen?pandidentificatie=0826100000000467&page=2"
+    En is attribuut _links.self.href gelijk aan "/adressen?pandidentificatie=0826100000000467"
+
+  Scenario: de zoekvraag levert meerdere pagina's en met de page parameter wordt een volgende pagina gevraagd
+    Als de request wordt gedaan naar /adressen?pandidentificatie=0826100000000467&page=3
+    Dan zitten er 20 adressen in het antwoord
+    En zijn dit adressen 41 tot en met 60 met deze pandidentificatie
+    En is attribuut _links.self.href gelijk aan "/adressen?pandidentificatie=0826100000000467&page=3"
+    En is attribuut _links.first.href gelijk aan "/adressen?pandidentificatie=0826100000000467&page=1"
+    En is attribuut _links.previous.href gelijk aan "/adressen?pandidentificatie=0826100000000467&page=2"
+    En is attribuut _links.next.href gelijk aan "/adressen?pandidentificatie=0826100000000467&page=4"
+
+  Scenario: de zoekvraag levert meerdere pagina's en met de page parameter wordt de laatste pagina gevraagd
+    Als de request wordt gedaan naar /adressen?pandidentificatie=0826100000000467&page=4
+    Dan zitten er 12 adressen in het antwoord
+    En zijn dit adressen 61 tot en met 72 met deze pandidentificatie
+    En is attribuut _links.first.href gelijk aan "/adressen?pandidentificatie=0826100000000467&page=1"
+    En is attribuut _links.previous.href gelijk aan "/adressen?pandidentificatie=0826100000000467&page=3"
+    En bevat het antwoord geen attribuut _links.next
+
+  Scenario: de zoekvraag levert meerdere pagina's en met de page parameter wordt een pagina bevraagd die niet bestaat
+    Als de request wordt gedaan naar /adressen?pandidentificatie=0826100000000467&page=15
```

Comment:
Als er een pagina wordt opgevraagd die niet bestaat, moet de provider een foutmelding teruggeven. Dit is een implementatiefout aan de client kant, die gedurende het development proces moet worden opgelost. De client moet een correct paginering mechanisme implementeren.
Hier kan het helpen om een last link te specificeren, want een client kan dan uit het antwoord halen hoeveel pagina's er zijn en zal dan geen pagina's opvragen die niet bestaan. Zie ook mijn andere opmerkingen.

Reply:
@MelvLee ben je het hiermee eens? Wil je bij het opgeven van een foute pagina (> laatste pagina) een foutmelding of de laatste pagina terugkrijgen?

Reply:
@strijm heb je hier last van in de implementatie

Reply:
Volgens mij is de laatste pagina teruggeven onlogisch. Je geeft een query-parameter op in een request (paginanummer) die geen resultaten oplevert. Normaal gesproken krijg je dan een lege collectie terug. De laatste pagina teruggeven zou dan gek zijn,maar ik weet niet wat best practice is.

Reply:
Volgens mij is de laatste pagina teruggeven onlogisch. Je geeft een query-parameter op in een request (paginanummer) die geen resultaten oplevert. Normaal gesproken krijg je dan een lege collectie terug. De laatste pagina teruggeven zou dan gek zijn,maar ik weet niet wat best practice is.

Reply:
Ik zou niet de laatste pagina teruggeven bij een paginanr groter dan de laatste pagina. Als we ervoor kiezen om een last page link toe te voegen, dan denk ik dat het logisher is om een foutmelding terug te geven bij een foute pagina

Reply:
opgelost. Pagina > last levert nu foutmelding op. Dit ook toegevoegd aan de foutafhandeling feature